### PR TITLE
Fix lazy evaluation of image names

### PR DESCRIPTION
**Pull request by darora**
Missed one occurrence in #165 ; added additional test for hitting this edge case

### CODE REVIEW

**Comment by gracew** [comment]
👍

File: src/main/groovy/com/palantir/gradle/docker/PalantirDockerPlugin.groovy [hunk 1]
```diff
@@ -142,7 +142,7 @@ class PalantirDockerPlugin implements Plugin<Project> {
 
             push.with {
                 workingDir dockerDir
-                commandLine 'docker', 'push', ext.name
+                commandLine 'docker', 'push', "${ -> ext.name}"
```
Comment (jmcampanini) [inline src/main/groovy/com/palantir/gradle/docker/PalantirDockerPlugin.groovy:145]: wat!? lol
